File: app/src/main/java/com/afterlogic/aurora/drive/presentation/modules/replace/view/ReplaceFileTypeFragment.java
package com.afterlogic.aurora.drive.presentation.modules.replace.view;

import android.arch.lifecycle.ViewModelProvider;
import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.afterlogic.aurora.drive.R;
import com.afterlogic.aurora.drive.presentation.common.binding.utils.UnbindableObservable;
import com.afterlogic.aurora.drive.presentation.common.modules.v3.view.BindingUtil;
import com.afterlogic.aurora.drive.presentation.common.modules.v3.view.InjectableMVVMFragment;
import com.afterlogic.aurora.drive.presentation.modules._baseFiles.v2.view.FileListArgs;
import com.afterlogic.aurora.drive.presentation.modules.replace.viewModel.ReplaceFileTypeViewModel;

/**
 * Created by aleksandrcikin on 04.07.17.
 * mail: mail@sunnydaydev.me
 */

public class ReplaceFileTypeFragment extends InjectableMVVMFragment<ReplaceFileTypeViewModel> {

    public static ReplaceFileTypeFragment newInstance(String type) {

        ReplaceFileTypeFragment fragment = new ReplaceFileTypeFragment();

        FileListArgs args = new FileListArgs();
        args.setType(type);

        fragment.setArguments(args.getBundle());

        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getViewModel().setArgs(new FileListArgs(getArguments()));
    }

    @Override
    public ViewDataBinding createBinding(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return DataBindingUtil.inflate(inflater, R.layout.replace_files_fragment, container, false);
    }

    @Override
    public ReplaceFileTypeViewModel createViewModel(ViewModelProvider provider) {
        return provider.get(ReplaceFileTypeViewModel.class);
    }

    @Override
    protected void bindStarted(ReplaceFileTypeViewModel vm, UnbindableObservable.Bag bag) {
        super.bindStarted(vm, bag);
        BindingUtil.bindProgressDialog(vm.progress, bag, getContext());
    }
}

File: app/src/main/java/com/afterlogic/aurora/drive/presentation/modules/replace/view/ReplaceFileTypeFragment.kt
package com.afterlogic.aurora.drive.presentation.modules.replace.view

import androidx.lifecycle.ViewModelProvider
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup

import com.afterlogic.aurora.drive.R
import com.afterlogic.aurora.drive.presentation.common.binding.utils.UnbindableObservable
import com.afterlogic.aurora.drive.presentation.common.modules.v3.view.BindingUtil
import com.afterlogic.aurora.drive.presentation.common.modules.v3.view.InjectableMVVMFragment
import com.afterlogic.aurora.drive.presentation.modules.baseFiles.v2.view.FileListArgs
import com.afterlogic.aurora.drive.presentation.modules.replace.viewModel.ReplaceFileListViewModel

/**
 * Created by aleksandrcikin on 04.07.17.
 * mail: mail@sunnydaydev.me
 */

class ReplaceFileTypeFragment : InjectableMVVMFragment<ReplaceFileListViewModel>() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel.setArgs(FileListArgs(arguments))
    }

    override fun createBinding(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): ViewDataBinding {
        return DataBindingUtil.inflate(inflater, R.layout.replace_files_fragment, container, false)
    }

    override fun createViewModel(provider: ViewModelProvider): ReplaceFileListViewModel {
        return provider.get(ReplaceFileListViewModel::class.java)
    }

    override fun bindStarted(vm: ReplaceFileListViewModel, bag: UnbindableObservable.Bag) {
        super.bindStarted(vm, bag)
        BindingUtil.bindProgressDialog(vm.progress, bag, context)
    }

    companion object {

        fun newInstance(type: String): ReplaceFileTypeFragment {

            val fragment = ReplaceFileTypeFragment()

            val args = FileListArgs()
            args.type = type

            fragment.arguments = args.bundle

            return fragment
        }
    }

}
